### DISCUSSION TITLE
Are there any plans to support XML parsing?

### DISCUSSION THREAD

**Discussion by ferrata** [darklang/dark#4607] [Q&A] [answered]
Hey folks! Sorry if the question is totally out of scope. I couldn't find a built-in way to parse an XML (maybe into XElement?). I assume it doesn't exist. 

Are there any plans to support it in the future?

Thanks!

**Comment by pbiggar** [comment, marked as answer]
Yes, absolutely. This would be built in user-space, and while there isn't super great support for that now (eg, no records or enums but you can do similar things with dicts) it should be possible to build them. While we don't plan to directly expose underlying xmlparsing tools, if any useful string manipulation functions are missing from the standard library, we'd be happy to add them (or take a PR to add them).
